src/main/scala/com/dwolla/cloudflare/AccountsClient.scala

```diff
@@ -0,0 +1,54 @@
+package com.dwolla.cloudflare
+
+import com.dwolla.cloudflare.common.JsonEntity._
+import com.dwolla.cloudflare.domain.model.accounts.Implicits._
+import com.dwolla.cloudflare.domain.dto.accounts.{AccountDTO, AccountMemberDTO, AccountRoleDTO, NewAccountMemberDTO}
+import com.dwolla.cloudflare.domain.model.accounts.{Account, AccountMember, AccountRole}
+import org.apache.http.client.methods.{HttpGet, HttpPost}
+import org.json4s.native._
+import org.json4s.{DefaultFormats, Formats}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AccountsClient(executor: CloudflareApiExecutor)(implicit val ec: ExecutionContext) {
```

Comment:
I wonder how much work it would be to extract an interface from this class that contains the various methods parameterized by effect type. (This would go towards eventually swapping out the HTTP client with something more functional, like http4s.)

e.g. something like

```scala
trait AccountsAPI[F[_]] {
  def listAccounts(): F[Set[Account]]
  def getById(accountId: String): F[Option[Account]]
}

class AccountsClient(executor: CloudflareApiExecutor)
                    (implicit ec: ExecutionContext) extends AccountsAPI[Future] {
  def listAccounts(): Future[Set[Account]] = ???
  def getById(accountId: String): Future[Option[Account]] = ???
}
```

Reply:
I will add this as a follow-up to investigate later.